ansible_collections/arista/avd/plugins/action/topology_generator.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+import glob
+from ansible.errors import AnsibleActionFail
+from ansible.plugins.action import ActionBase
+import ansible_collections.arista.avd.plugins.plugin_utils.topology_generator_utils as gt
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+        if self._task.args and "structured_config" not in self._task.args:
+            raise AnsibleActionFail("Missing 'structured_config' variable.")
+        path = self._task.args["structured_config"]
+        self.driver_func(path)
+        return result
+
+
+    def driver_func(self, directory_path):
+        graph_dict = {}
+        json_input = []
+        finaldict = {}
+        files = glob.glob(directory_path+"/*.yml")
+
+        output_list = []
+        for index, file in enumerate(files):
+            data = gt.read_yaml_file(file)
+            node_dict = gt.create_node_dict(data, file)
+            output_list = gt.structured_config_to_topology_input(
+                output_list, node_dict, data['diagram_groups'], current_dict={})
+        root_dict = gt.find_root_nodes(output_list[0])
+        nodes = []
+        output_list[0]["nodes"].append(root_dict)
+        global_node_list, graph_dict = gt.create_graph_dict(output_list)
+        nodes_rank_dict = gt.find_node_levels(graph_dict, '0', global_node_list)
+        rank_nodes_list = []
+        for k, v in nodes_rank_dict.items():
+            rank_nodes_list = rank_nodes_list + v
+        undefined_rank_nodes = list(set(rank_nodes_list) - set(global_node_list)) + \
```

Comment:
don't use `\` to continue line - cf PEP8 https://peps.python.org/pep-0008/#maximum-line-length

plus it looks like you are doing with 2 sets A and B
```
res = list( A-B) + list (B-A) 
```

which is "the list of the element in A but not in B concatenated with the list of the element in B but not in A" a.k.a symmetric difference which can be obtained in a simpler fashion in python:
https://docs.python.org/3/tutorial/datastructures.html#sets

```
undefined_rank_nodes = list(set(rank_nodes_list) ^ set(global_node_list))
```